portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java

```diff
@@ -24,35 +24,70 @@
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
+import com.google.sps.data.Text;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.util.List;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/jsonData")
-public class DataServlet2 extends HttpServlet {
-    private ArrayList<String> comments = new ArrayList();
+
+public class DataServlet2 extends HttpServlet {  
+  // This an array List that will be used in DeletedataSerlet.java to delete all comments
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = convertToJson(comments);
+      Query query = new Query("Text").addSort("timestamp", SortDirection.DESCENDING);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      PreparedQuery results = datastore.prepare(query);
+      List<Text> texts = new ArrayList<>();
 
-    response.setContentType("application/json;");
-    response.getWriter().println(json);
-  }
+      String maxCommentsString = request.getParameter("maxCount");
+      int maxComment;
+      try {
+        maxComment = Integer.parseInt(maxCommentsString);
+      } catch (NumberFormatException e) {
+        System.err.println("Could not convert to int: " + maxCommentsString);
+        maxComment = -1;
+      }
 
-  private String convertToJson(ArrayList<String> comments){
-      String json = "{";
-      for(int i = 0; i < comments.size(); ++i){
-          if (i != 0){
-              json += ", ";
-          }
-          json += "\"Comment " + (i + 1) + "\": ";
-          json += "\"" + comments.get(i) + "\"";
+      // Check that the input is between 0 and 50
+      if (maxComment < 0 || maxComment > 50) {
+        System.err.println("User choice is out of range: " + maxCommentsString);
+        maxComment = -1;
       }
-      json += "}";
-      return json;
+      
+      int currentComment = 0;
+      for (Entity entity : results.asIterable()) {
+        if(maxComment != -1){
```

Comment:
If `maxComment` does equal -1, return an error in the request with `response.sendError`